CMakeLists.txt

```diff
@@ -1,15 +1,27 @@
 cmake_minimum_required(VERSION 3.16)
 
+# Used as a tool to make sure that the standard used by tools
+# and by the projects cannot get out of sync
+set(USER_CXX_STANDARD 20)
```

Comment:
Why are we adding another standard? Isn't CMAKE_CXX_STANDARD enough?

Reply:
Because that was 2 values to keep track of (see how USER_CXX_STANDARD is actually use) and it was possible/easy for them to get out of sync.

Reply:
This line adds another line of code for standard. I would remove this and use `CMAKE_CXX_STANDARD` everywhere to set the standard.